src/main/java/edu/kit/compiler/transform/IRExpressionVisitor.java

```diff
@@ -0,0 +1,132 @@
+
+    private final TransformContext context;
+
+    public IRExpressionVisitor(TransformContext context) {
+        this.context = context;
+    }
+
+    @Override
+    public Node visit(ExpressionNode.BinaryExpressionNode binaryExpressionNode) {
+        return null;
+    }
+
+    @Override
+    public Node visit(ExpressionNode.UnaryExpressionNode unaryExpressionNode) {
+        return null;
+    }
+
+    @Override
+    public Node visit(ExpressionNode.MethodInvocationExpressionNode methodInvocationExpressionNode) {
+        return null;
+    }
+
+    @Override
+    public Node visit(ExpressionNode.FieldAccessExpressionNode fieldAccessExpressionNode) {
+        return null;
+    }
+
+    @Override
+    public Node visit(ExpressionNode.ArrayAccessExpressionNode arrayAccessExpressionNode) {
+        return null;
+    }
+
+    @Override
+    public Node visit(ExpressionNode.IdentifierExpressionNode identifierExpressionNode) {
+        Construction con = context.getConstruction();
+        if (identifierExpressionNode.getDefinition().getKind() == DefinitionKind.LocalVariable) {
+            Mode mode = getMode(identifierExpressionNode.getResultType());
+            return con.getVariable(context.getVariableIndex(identifierExpressionNode.getIdentifier()), mode);
+        } else if (identifierExpressionNode.getDefinition().getKind() == DefinitionKind.Parameter) {
+            return context.createParamNode(identifierExpressionNode.getIdentifier());
+        } else {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    @Override
+    public Node visit(ExpressionNode.ThisExpressionNode thisExpressionNode) {
+        return null;
+    }
+
+    @Override
+    public Node visit(ExpressionNode.ValueExpressionNode valueExpressionNode) {
+        // determine type of value expression and value and set the expression
+        // handle accordingly
+
+        // the new node that should be craeted here
+        Node n = null;
+        switch (valueExpressionNode.getValueType()) {
+            case Null: {
+                n = handleNullExpressionNode(valueExpressionNode);
+                break;
+            }
+            case False:
+            case True: {
+                throw new UnsupportedOperationException();
+            }
+            case IntegerLiteral: {
+                n = handleIntegerExpressionNode(valueExpressionNode);
+                break;
+            }
+        }
+
+        // add node to current construction
+        return n;
+    }
+
+    private Node handleIntegerExpressionNode(ExpressionNode.ValueExpressionNode valueExpressionNode) {
+        if (valueExpressionNode.getValueType() != IntegerLiteral) {
+            throw new IllegalArgumentException("can only evaluate integer expressions");
+        }
+
+        if (valueExpressionNode.getLiteralValue().isEmpty()) {
+            throw new IllegalArgumentException("cannot get empty integer literal");
```

Comment:
All of these checks have already been done during semantic analysis. We can assume that the program (i.e. the AST) during the transformation phase.